### PR TITLE
Factor out CoreConfig

### PR DESCRIPTION
**Pull request by mrjrieke**

### CODE REVIEW

**Review by KarnveerSGill — APPROVED** [review]

File: pkg/trcx/xutil/xmanager.go [hunk 1]
```diff
@@ -479,31 +480,31 @@ func GenerateSeedSectionFromVaultRaw(config *eUtils.DriverConfig, templateFromVa
 				goMod.Release()
 			}
 			if errSeed != nil {
-				eUtils.LogAndSafeExit(config, errSeed.Error(), -1)
+				eUtils.LogAndSafeExit(&c.CoreConfig, errSeed.Error(), -1)
```
Comment (joel-rieke) [inline pkg/trcx/xutil/xmanager.go:483]: This doesn't look right.

File: pkg/utils/driverconfig.go [hunk 2]
```diff
@@ -149,7 +146,7 @@ func ConfigControl(ctx ProcessContext, configCtx *ConfigContext, config *DriverC
 			for _, projectFile := range projectFiles {
 				projectStartDir := config.StartDir[0]
 
-				if projectFile.Name() != "Common" && config.WantCerts && config.WantKeystore == "" {
+				if projectFile.Name() != "Common" && config.CoreConfig.WantCerts && config.WantKeystore == "" {
```
Comment (joel-rieke) [inline pkg/utils/driverconfig.go:149]: This should be driverConfig now.

File: pkg/utils/vaultUtil.go [hunk 3]
```diff
@@ -9,50 +9,51 @@ import (
...
 	"github.com/trimble-oss/tierceron/buildopts/coreopts"
+	"github.com/trimble-oss/tierceron/pkg/core"
 	helperkv "github.com/trimble-oss/tierceron/pkg/vaulthelper/kv"
 	sys "github.com/trimble-oss/tierceron/pkg/vaulthelper/system"
 )
 
 // Helper to easiliy intialize a vault and a mod all at once.
 func InitVaultMod(config *DriverConfig) (*DriverConfig, *helperkv.Modifier, *sys.Vault, error) {
-	LogInfo(config, "InitVaultMod begins..")
+	LogInfo(&config.CoreConfig, "InitVaultMod begins..")
```
Comment (joel-rieke) [inline pkg/utils/vaultUtil.go:19]: This should be driverConfig now.

File: atrium/vestibulum/trcsh/deployutil/deployutil.go [hunk 4]
```diff
@@ -75,14 +75,14 @@ func LoadPluginDeploymentScript(config *eUtils.DriverConfig, trcshConfig *capaut
...
 
 	// Swapping in project root...
-	configRoleSlice := strings.Split(config.AppRoleConfig, ":")
-	mergedEnvRaw := config.EnvRaw
-	tokenName := "config_token_" + config.Env
+	configRoleSlice := strings.Split(driverConfig.AppRoleConfig, ":")
+	mergedEnvRaw := driverConfig.EnvRaw
+	tokenName := "config_token_" + driverConfig.Env
 	readToken := ""
-	autoErr := eUtils.AutoAuth(config, &configRoleSlice[1], &configRoleSlice[0], &readToken, &tokenName, &config.Env, &config.VaultAddress, &mergedEnvRaw, "config.yml", false)
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply atrium/vestibulum/trcsh/deployutil/deployutil.go:85]: Reverted to config.yml.

File: pkg/cli/trcconfigbase/trcconfig.go [hunk 5]
```diff
@@ -446,35 +455,32 @@ func CommonMain(envPtr *string,
...
+			OutputMemCache:    driverConfigBase.OutputMemCache,
+			MemFs:             driverConfigBase.MemFs,
 			Diff:              *diffPtr,
 			FileFilter:        fileFilterSlice,
 			VersionInfo:       eUtils.VersionHelper,
 		}
 
-		if len(configBase.DeploymentConfig) > 0 {
-			config.DeploymentConfig = configBase.DeploymentConfig
+		if len(driverConfig.DeploymentConfig) > 0 {
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcconfigbase/trcconfig.go]: Reverted to logic it was before.

File: pkg/utils/vaultUtil.go (hunk 3 above)
Comment (mrjrieke) [reply pkg/utils/vaultUtil.go]: Changed to driverConfig *DriverConfig

File: pkg/cli/trcpubbase/trcpub.go [hunk 6]
```diff
@@ -54,60 +55,67 @@ func CommonMain(envPtr *string,
...
 			*logFilePtr = "./" + coreopts.BuildOptions.GetFolderPrefix(nil) + "pub.log"
 		}
 		f, err := os.OpenFile(*logFilePtr, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
 		logger := log.New(f, "[INIT]", log.LstdFlags)
-		configBase = &eUtils.DriverConfig{Insecure: true, Log: logger, ExitOnFailure: true}
-		eUtils.CheckError(configBase, err, true)
+		driverConfigBase = &eUtils.DriverConfig{
+			CoreConfig: core.CoreConfig{
+				Log:           logger,
+				ExitOnFailure: true,
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcpubbase/trcpub.go:78]: Log is now last for consistency.

File: pkg/cli/trcsubbase/trcsub.go [hunk 7]
```diff
@@ -85,18 +86,22 @@ func CommonMain(envPtr *string, addrPtr *string, envCtxPtr *string,
...
 
 		logger := log.New(f, "[INIT]", log.LstdFlags)
-		configBase = &eUtils.DriverConfig{Insecure: *insecurePtr,
-			EndDir:        *endDirPtr,
-			Log:           logger,
-			ExitOnFailure: exitOnFailure}
+		configBase = &eUtils.DriverConfig{
+			CoreConfig: core.CoreConfig{
+				Log:           logger,
+				ExitOnFailure: exitOnFailure,
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcsubbase/trcsub.go:91]: Log is now last for consistency.

File: pkg/cli/trcxbase/trcxbase.go [hunk 8]
```diff
@@ -445,28 +475,30 @@ skipDiff:
...
-						Diff:              *diffPtr,
-						Update:            messenger,
-						VersionInfo:       eUtils.VersionHelper,
-						DynamicPathFilter: pGen,
-						SubPathFilter:     strings.Split(pGen, ","),
-						FileFilter:        fileFilter,
-						ExitOnFailure:     true,
-						Trcxr:             *readOnlyPtr,
+						CoreConfig: core.CoreConfig{
+							WantCerts:         *wantCertsPtr,
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcxbase/trcxbase.go:479]: Log is now last for consistency.

File: pkg/utils/driverconfig.go (hunk 2 above)
Comment (mrjrieke) [reply pkg/utils/driverconfig.go]: Latest checkin should address this now.

File: pkg/trcx/xutil/xmanager.go (hunk 1 above)
Comment (mrjrieke) [reply pkg/trcx/xutil/xmanager.go]: Should be fixed now.

File: pkg/cli/trcsubbase/trcsub.go [hunk 9]
```diff
@@ -85,18 +86,22 @@ func CommonMain(envPtr *string, addrPtr *string, envCtxPtr *string,
 		}
 
 		logger := log.New(f, "[INIT]", log.LstdFlags)
-		configBase = &eUtils.DriverConfig{Insecure: *insecurePtr,
-			EndDir:        *endDirPtr,
-			Log:           logger,
-			ExitOnFailure: exitOnFailure}
+		configBase = &eUtils.DriverConfig{
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcsubbase/trcsub.go]: Should show as fixed now.

File: pkg/cli/trcxbase/trcxbase.go [hunk 10]
```diff
@@ -413,9 +425,15 @@ skipDiff:
 				}
 				if !*noVaultPtr && *tokenPtr == "" {
 					//Ask vault for list of dev.<id>.* environments, add to envSlice
-					authErr := eUtils.AutoAuth(&eUtils.DriverConfig{Insecure: *insecurePtr, Log: logger, ExitOnFailure: true}, secretIDPtr, appRoleIDPtr, tokenPtr, tokenNamePtr, &baseEnv, addrPtr, envCtxPtr, "", *pingPtr)
+					authErr := eUtils.AutoAuth(&eUtils.DriverConfig{
+						CoreConfig: core.CoreConfig{
+							Log:           logger,
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcxbase/trcxbase.go]: Fixed now.

File: pkg/cli/trcxbase/trcxbase.go [hunk 11]
```diff
@@ -427,12 +445,24 @@ skipDiff:
 			recursivePathBuilder = func(testMod *kv.Modifier, pGen string, dynamicPathParts []string) {
 				if len(dynamicPathParts) == 0 {
 					if !*noVaultPtr && *tokenPtr == "" {
-						authErr := eUtils.AutoAuth(&eUtils.DriverConfig{Insecure: *insecurePtr, Log: logger, ExitOnFailure: true}, secretIDPtr, appRoleIDPtr, tokenPtr, tokenNamePtr, envPtr, addrPtr, envCtxPtr, "", *pingPtr)
+						authErr := eUtils.AutoAuth(&eUtils.DriverConfig{
+							CoreConfig: core.CoreConfig{
+								Log:           logger,
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcxbase/trcxbase.go]: This should be fixed now.

File: pkg/cli/trcxbase/trcxbase.go [hunk 12]
```diff
@@ -675,12 +712,23 @@ skipDiff:
 			for _, section := range sectionSlice {
 				var servicesWanted []string
 				if !*noVaultPtr && *tokenPtr == "" {
-					authErr := eUtils.AutoAuth(&eUtils.DriverConfig{Insecure: *insecurePtr, Log: logger, ExitOnFailure: true}, secretIDPtr, appRoleIDPtr, tokenPtr, tokenNamePtr, envPtr, addrPtr, envCtxPtr, "", *pingPtr)
+					authErr := eUtils.AutoAuth(&eUtils.DriverConfig{
+						CoreConfig: core.CoreConfig{
+							Log:           logger,
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcxbase/trcxbase.go]: Should be fixed now.

File: pkg/cli/trcxbase/trcxbase.go [hunk 13]
```diff
@@ -675,12 +712,23 @@ skipDiff:
...
+							ExitOnFailure: true,
+						},
+						Insecure: *insecurePtr}, secretIDPtr, appRoleIDPtr, tokenPtr, tokenNamePtr, envPtr, addrPtr, envCtxPtr, "", *pingPtr)
 					if authErr != nil {
 						// Retry once.
-						authErr := eUtils.AutoAuth(&eUtils.DriverConfig{Insecure: *insecurePtr, Log: logger, ExitOnFailure: true}, secretIDPtr, appRoleIDPtr, tokenPtr, tokenNamePtr, envPtr, addrPtr, envCtxPtr, "", *pingPtr)
+						authErr := eUtils.AutoAuth(&eUtils.DriverConfig{
+							CoreConfig: core.CoreConfig{
+								Log:           logger,
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcxbase/trcxbase.go]: Should be fixed now.

File: pkg/cli/trcxbase/trcxbase.go [hunk 14]
```diff
@@ -703,36 +751,38 @@ skipDiff:
...
-					FileFilter:        fileFilter,
-					SubPathFilter:     strings.Split(*eUtils.SubPathFilter, ","),
-					ProjectSections:   configCtx.ProjectSectionsSlice,
-					ServiceFilter:     serviceFilterSlice,
-					ExitOnFailure:     true,
-					Trcxe:             trcxeList,
-					Trcxr:             *readOnlyPtr,
+					CoreConfig: core.CoreConfig{
+						WantCerts:         *wantCertsPtr,
+						Log:               logger,
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/cli/trcxbase/trcxbase.go]: Should be fixed now.

File: pkg/core/util/util.go [hunk 15]
```diff
@@ -143,7 +144,7 @@ func LoadBaseTemplate(config *eUtils.DriverConfig, templateResult *extract.Templ
 	if goMod != nil {
 		cds = new(vcutils.ConfigDataStore)
 		goMod.Version = goMod.Version + "***X-Mode"
-		cds.Init(config, goMod, true, true, project, commonPaths, service) //CommonPaths = "" - empty - not needed for tenant config
+		cds.Init(&config.CoreConfig, goMod, true, true, project, commonPaths, service) //CommonPaths = "" - empty - not needed for tenant config
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/core/util/util.go]: Diff looks out of date... this variable has been renamed to driverCOnfig.

File: pkg/core/util/util.go [hunk 16]
```diff
@@ -220,9 +221,9 @@ func SeedVaultById(config *eUtils.DriverConfig, goMod *helperkv.Modifier, servic
 	sliceValueSection = append(sliceValueSection, templateResult.ValueSection)
 	sliceSecretSection = append(sliceSecretSection, templateResult.SecretSection)
 
-	xutil.CombineSection(config, sliceTemplateSection, maxDepth, templateCombinedSection)
-	xutil.CombineSection(config, sliceValueSection, -1, valueCombinedSection)
-	xutil.CombineSection(config, sliceSecretSection, -1, secretCombinedSection)
+	xutil.CombineSection(&config.CoreConfig, sliceTemplateSection, maxDepth, templateCombinedSection)
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/core/util/util.go]: Diff looks out of date.

File: pkg/core/util/util.go [hunk 17]
```diff
@@ -247,23 +248,23 @@ func SeedVaultById(config *eUtils.DriverConfig, goMod *helperkv.Modifier, servic
 	//VaultInit Section Begins
 	if strings.Contains(indexPath, "/PublicIndex/") {
 		config.ServicesWanted = []string{""}
-		config.WantCerts = false
+		config.CoreConfig.WantCerts = false
```
[the remark this answers is not in the document]
Comment (mrjrieke) [reply pkg/core/util/util.go]: diff looks out of date.

File: atrium/vestibulum/trcsh/deployutil/deployutil.go [hunk 18]
```diff
@@ -14,30 +14,30 @@ import (
...
 			}
-			mergedEnvRaw := config.EnvRaw
+			mergedEnvRaw := driverConfig.EnvRaw
 			// Swapping in project root...
 			configRoleSlice := strings.Split(*trcshConfig.ConfigRole, ":")
-			tokenName := "config_token_" + config.EnvRaw
+			tokenName := "config_token_" + driverConfig.EnvRaw
 			readToken := ""
-			autoErr := eUtils.AutoAuth(config, &configRoleSlice[1], &configRoleSlice[0], &readToken, &tokenName, &config.Env, &config.VaultAddress, &mergedEnvRaw, "config.yml", false)
+			autoErr := eUtils.AutoAuth(driverConfig, &configRoleSlice[1], &configRoleSlice[0], &readToken, &tokenName, &driverConfig.Env, &driverConfig.VaultAddress, &mergedEnvRaw, "driverConfig.yml", false)
```
Comment (joel-rieke) [inline atrium/vestibulum/trcsh/deployutil/deployutil.go:28]: This should just by config.yml

**Pull request merged**